### PR TITLE
ComplexCommand transformations visitor

### PR DESCRIPTION
**Pull request by DamianWdowiak**
Task [A visitor for a transformation of a complex command](https://trello.com/c/iAlTC2HV/24-a-visitor-for-a-transformation-of-a-complex-command)

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/command/visitor/TransformationsVisitor.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package edu.kis.powp.jobs2d.command.visitor;
+
+import edu.kis.powp.jobs2d.command.*;
+import edu.kis.powp.jobs2d.drivers.transformation.Point;
+import edu.kis.powp.jobs2d.drivers.transformation.PointTransformation;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/visitor/TransformationsVisitor.java:5]: it seems that the whole transformation package is missing. I cannot find classes Point, Rotate etc.

File: src/main/java/edu/kis/powp/jobs2d/command/ICompoundCommand.java [hunk 2]
```diff
@@ -13,6 +13,6 @@
 
     @Override
     default void accept(Visitor visitor) {
-        visitor.visitICompoundCommand(this);
+        visitor.visit(this);
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/ICompoundCommand.java]: Please do not rename existing code while working on a feature. Such refactoring is not a necessity and should be done as separate commit and PR.

**Review by kowallus — CHANGES_REQUESTED** [review]
Some files are missing

File: src/main/java/edu/kis/powp/jobs2d/command/visitor/TransformationsVisitor.java (hunk 1 above)
Comment (DamianWdowiak) [reply src/main/java/edu/kis/powp/jobs2d/command/visitor/TransformationsVisitor.java:5]: Files are already on `iis-powp-2021:io2-dev`

File: src/test/java/edu/kis/powp/jobs2d/events/SelectRotateCurrentCommandOptionListener.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import edu.kis.powp.jobs2d.features.CommandsFeature;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class SelectRotateCurrentCommandOptionListener implements ActionListener {
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        ICompoundCommand command = (ICompoundCommand) CommandsFeature.getDriverCommandManager().getCurrentCommand();
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/SelectRotateCurrentCommandOptionListener.java]: Casting is not necessary with visitor!
Please try to remove dependency to CommandsFeature (it would help with future refactoring)

File: src/test/java/edu/kis/powp/jobs2d/events/SelectRotateCurrentCommandOptionListener.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        TransformationsVisitor transformationsVisitor = new TransformationsVisitor(new Rotate(Math.PI / 2));
+        transformationsVisitor.visitICompoundCommand(command);
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/SelectRotateCurrentCommandOptionListener.java]: visitors should be called using accept!

File: src/main/java/edu/kis/powp/jobs2d/command/visitor/TransformationsVisitor.java (hunk 1 above)
Comment (kowallus) [reply src/main/java/edu/kis/powp/jobs2d/command/visitor/TransformationsVisitor.java:5]: OK. Sorry.

**Review by kowallus — CHANGES_REQUESTED** [review]
Please fix the test.
